imports/lib/ValidateShape.ts

```diff
@@ -0,0 +1,6 @@
+type ValidateShape<T, Shape> =
+    T extends Shape ?
```

Comment:
Quick note that this was previously `T extends Partial<Shape>`. I'm not sure why given that I was already passing in `doc: ValidateShape<U, Partial<T>>` in `Base`'s declaration of `insert`, but it's clearly not what we want for method declarations (partial types are not valid)